### ISSUE TITLE
Remove `id_source` parameter for ui.indent

### ISSUE THREAD

**Issue by FreePhoenix888** [emilk/egui#4066]
**Is your feature request related to a problem? Please describe.**

We have to write id_source without a reason when creating an indent. Some of developers have added TODO to code to remove it (with question)
https://github.com/emilk/egui/blob/ca8c8792c2a6a65656105b971998f937edcb836d/crates/egui/src/ui.rs#L1775-L1780

**Describe the solution you'd like**

Remove it. Is there any reason to have it?

**Describe alternatives you've considered**

Do not remove it for compability reasons 🤔 

**Additional context**

**Comment by emilk** [comment]
Yeah, it doesn't really belong there since the ident area doesn't store any state. Feel free to make a PR to remove it. Breaking changes are ok!
